agent/bench-scripts/pbench-fio

```diff
@@ -598,19 +609,9 @@ function fio_run_job() {
 	echo "running fio job: $fio_job_file ($(basename $benchmark_results_dir))"
 
 	if [ ! -z "${clients}" ]; then
-		debug_log "creating directories on the clients"
+		debug_log "killing any old fio processes, creating directories, and starting new a fio process on the clients"
 		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} mkdir -p ${benchmark_results_dir} &
-		done
-		wait
-		debug_log "killing any old fio process on the clients"
-		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} "killall fio >/dev/null 2>&1" &
-		done
-		wait
-		debug_log "starting new fio process on the clients"
-		for client in "${client_names[@]}"; do
-			ssh ${ssh_opts} ${client} "cd ${benchmark_results_dir} && screen -dmS fio-server bash -c '$benchmark_bin --server=,${client_ports[${client}]} 2>&1 >client-result.txt'"
+			ssh ${ssh_opts} ${client} "killall ${benchmark} >/dev/null 2>&1; mkdir -p ${benchmark_results_dir} && cd ${benchmark_results_dir} && screen -dmS fio-server bash -c '${benchmark_bin} --server=,${client_ports[${client}]} 2>&1 >client-result.txt'"
```

Comment:
I believe that is a poor way of expressing it.  I'll make it clear.

Reply:
I think I was hoping for a yes-or-no answer.  😁

You've changed the code so that it is now including `stderr` with the rest of `stdout` in `client-result.txt`...are we sure that _that_ is what we want?  (I have no idea what the respective contents of `stderr`, `stdout`, and the "results" of the benchmark are, but I think we already have an issue from Hifza which boils down to having `stderr` output mingled with benchmark results-output....)

Reply:
I am not following how I changed that, as in both cases in the diff above we have `$benchmark_bin ... 2>&1 >client-result.txt`.

What I think you are referring to is the latest change, not shown above, where I capture both `stderr` and `stdout` to the `client-result.txt` file.

This is intentional.  The original code, because it is run under `screen`, consumes the `stdout` entirely.  So any output from `stderr` is lost entirely.

I tried that with the following:

```
$ cat test.sh; screen -dmS test-foo bash -c './test.sh 2>&1 > test.sh.out'; cat test.sh.out
#!/bin/bash

echo "this is written to stdout"
echo "this is written to stderr" >&2
exit 0
this is written to stdout
```

By redirecting `stderr` to `stdout` after `stdout` has the output file in place gives us a chance to capture the `stderr` output as well.

Reply:
I think we agree that losing `stderr` would be a Bad Thing.  However, mixing `stderr` text into `stdout` text such that the benchmark post-processing does something goofy would also be a Bad Thing.

I just want to confirm that this isn't going to break anything.  (I'm _sure_ that Hifza reported a case where something was dumping an error into the top of the benchmark output file and this was causing post-processing to fail...but I cannot find it, now. 🙁)  Is `client-result.txt` subject to post-processing?  If so, is that stage going to do the right thing if there is `stderr` output mixed in?

Reply:
Updated to use a separate `client-result.err` file.